### PR TITLE
Additional http methods

### LINKED ISSUE #1339: Additional HTTP methods support
Vert.x 2 had support for any extended HTTP method as methods are accepted as String. Vert.x has now limited itself to work only with standard HTTP methods and this has been a limitation for our application to migrate to Vert.x 3.

As per the [discussion](https://groups.google.com/forum/#!msg/vertx/PoJMC_wEsNo/-k3hE1N_AwAJ) with @vietj , it is expected in 3.3. Adding this issue for tracking convenience.

### PR DESCRIPTION
**Pull request by vietj**
fixes #1339

please review

### CODE REVIEW

File: src/main/java/io/vertx/core/http/impl/HttpClientRequestPushPromise.java [hunk 1]
```diff
@@ -116,6 +117,16 @@ public HttpMethod method() {
   }
 
   @Override
+  public String getRawMethod() {
+    throw new UnsupportedOperationException("todo");
```
Comment (cescoffier) [inline src/main/java/io/vertx/core/http/impl/HttpClientRequestPushPromise.java]: Write todo the same "way" in both places.

Comment (cescoffier) [inline src/main/java/io/vertx/core/http/impl/HttpClientRequestPushPromise.java]: Is it a TODO or a not supported feature.

**Comment by cescoffier** [comment]
I think it should be documented somewhere.

File: src/main/java/io/vertx/core/http/impl/HttpClientRequestPushPromise.java (hunk 1 above)
Comment (vietj) [inline src/main/java/io/vertx/core/http/impl/HttpClientRequestPushPromise.java]: oups

**Comment by vietj** [comment]
fair enough

**Pull request merged**